### DISCUSSION TITLE
[Feature Request]: Allow setting of ATAK team color and role

### DISCUSSION THREAD

**Discussion by RichieFrame** [meshtastic/firmware#4787] [Ideas]
### Platform

Cross-Platform

### Description

Currently the tak packet is built with a hardcoded value for these, here: [PositionModule.cpp](https://github.com/meshtastic/firmware/blob/0f3450ad441f766bd451e9f2d2cfea09b5730431/src/modules/PositionModule.cpp#L278)

The protobufs already contain the other options for team colors and roles, please allow setting these for TAK_TRACKER devices

**Comment by gitjusnix** [comment]
Seconded. Need ability to assign Team Colors to ATAK PLI of meshtastic nodes, whether their role is TAK or TAK_Tracker. Please; default color of Cyan is for rookies haha.

**Comment by skadakar** [comment]
Even just the icon type would be useful, if its a person, a team or a vehicle would be nice to have matching the mil-std 2525

**Comment by RichieFrame** [comment]
So just a heads up on where I was with this, I had test code for BaseUI control over both the role and team color around the 2.7.11 days, but then at some point the code stopped working (UI menu would not even load), and now even if I hardcode the enum for a different color and compile the firmware it does not seem to broadcast it, which is highly suspicious and I for sure need to look into why, my older nodes with the working code still broadcast the selected color so I know the issue is on the transmit end not the receive end (most likely)

I had planned to post a PR for the initial part of the code after the UI menu refactor was complete but this stopped working before then, once I have figured out where the issue is I will try again
